cpu/fe310/cpu.c

```diff
@@ -105,6 +109,17 @@ void cpu_init(void)
     flash_init();
 #endif
 
+#ifdef MODULE_PMP_STACK_GUARD
+    /* FLASH ROM */
+    write_pmpaddr(0, make_napot(0x20000000, 0x20000000));
+    set_pmpcfg(0, PMP_L | PMP_NAPOT | PMP_X | PMP_W | PMP_R);
+
+    /* allow R/W for On-Chip Peripherals */
+    write_pmpaddr(1, 0x02000000);
+    write_pmpaddr(2, 0x20000000);
+    set_pmpcfg(2, PMP_L | PMP_TOR | PMP_W | PMP_R);
+#endif
+
```

Comment:
Not happy with this code block. Instead of 'stack guarding' being generic RISC-V + PMP, it now has dependencies with the specific CPU and its address map. Needs rework in an additional PR? Does RIOT expose the memory map of each cpu? I remember the `CPU_RAM_BASE` do be a hassle from my last PR...